portfolio/src/main/java/com/google/sps/servlets/CommentServlet.java

```diff
@@ -53,21 +55,20 @@ private void writeToDatabase(Entity toWrite) {
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-
-    List<Comment> comments = getComments();
+    int commentLimit = Integer.parseInt(request.getParameter("number-comments"));
```

Comment:
It might help to do some validation on the request before using the values.

1. getParameter() returns null if the query parameter doesn't exist. This might cause an exception here.
2. What happens if number-comments is set to a non-numeric value (e.g. "abc")?
3. Are negative / zero comment values handled properly?

Reply:
I added an if statement before getComments() in the java file to validate that the input is one of the choices (5, 10, 15, 50, 100). I also validate that it isn't null in the .js file, but this will be good as a double precaution. Thank you!

Reply:
Update: I switched it to use the getParameter method that I already had, which checks for empty string and null. I don't think there will ever be an issue about negative numbers because the only choices for the select are 5, 10, 15, 50, and 100. Let me know, thanks.